core/src/test/kotlin/com/segment/analytics/kotlin/core/JSONTests.kt

```diff
@@ -3,503 +3,712 @@ package com.segment.analytics.kotlin.core
+            val jsonObject = JsonObject(mapOf("keyed" to JsonPrimitive(18)))
 
-        try {
-            val keyedValue = jsonObject.getString("keyed")
-            assertTrue(keyedValue.equals("18"))
-        } catch (e: Exception) {
-            fail("Should not throw when not int primitive")
+            try {
+                val keyedValue = jsonObject.getString("keyed")
+                assertTrue(keyedValue.equals("18"))
+            } catch (e: Exception) {
+                fail("Should not throw when not int primitive")
+            }
         }
-    }
 
-    @Test
-    fun `get string optional`() {
-        val jsonObject = JsonObject(mapOf("keyed" to JsonPrimitive(null as String?)))
+        @Test
+        fun `get string optional`() {
+            val jsonObject = JsonObject(mapOf("keyed" to JsonPrimitive(null as String?)))
 
-        try {
-            val keyedValue = jsonObject.getString("keyed")
-            assertTrue(keyedValue == null)
-        } catch (e: Exception) {
-            fail("Should not throw when null string")
+            try {
+                val keyedValue = jsonObject.getString("keyed")
+                assertTrue(keyedValue == null)
+            } catch (e: Exception) {
+                fail("Should not throw when null string")
+            }
         }
-    }
 
-    @Test
-    fun `get normal integer succeeds`() {
-        val jsonObject = JsonObject(mapOf("keyed" to JsonPrimitive(18)))
+        @Test
+        fun `get normal integer succeeds`() {
+            val jsonObject = JsonObject(mapOf("keyed" to JsonPrimitive(18)))
 
-        val keyedValue = jsonObject.getInt("keyed")
+            val keyedValue = jsonObject.getInt("keyed")
 
-        assertTrue(keyedValue?.equals(18) ?: false)
-    }
+            assertTrue(keyedValue?.equals(18) ?: false)
+        }
 
-    @Test
-    fun `get normal integer fails`() {
-        val jsonObject = JsonObject(mapOf("keyed" to JsonPrimitive(19)))
+        @Test
+        fun `get normal integer fails`() {
+            val jsonObject = JsonObject(mapOf("keyed" to JsonPrimitive(19)))
 
-        val keyedValue = jsonObject.getInt("keyed")
+            val keyedValue = jsonObject.getInt("keyed")
 
-        assertFalse(keyedValue?.equals(18) ?: true)
-    }
+            assertFalse(keyedValue?.equals(18) ?: true)
+        }
 
-    @Test
-    fun `get integer bad value throws`() {
+        @Test
+        fun `get integer bad value throws`() {
 
-        val jsonObject = JsonObject(mapOf("keyed" to JsonPrimitive("18")))
+            val jsonObject = JsonObject(mapOf("keyed" to JsonPrimitive("18")))
 
-        try {
-            val keyedValue = jsonObject.getInt("keyed")
-            assertTrue(keyedValue?.equals(18) ?: false)
-        } catch (e: Exception) {
-            fail("Should not throw when not int primitive")
+            try {
+                val keyedValue = jsonObject.getInt("keyed")
+                assertTrue(keyedValue?.equals(18) ?: false)
```

Comment:
nit: it's probably better to use `assertEquals`. same applies to other similar usages in this file

Reply:
yup good catch, i copied the language from other parts. will update everywhere :)